src/thread_manager.ts

```diff
@@ -76,6 +77,11 @@ export class ThreadManager<SCG extends ExtendableGenerics = DefaultGenerics> {
     return threadsById;
   }
 
+  public resetState = () => {
+    this.unregisterSubscriptions();
```

Comment:
```suggestion
    this.unregisterSubscriptions();
```
Hmmm - good point but now I'm conflicted. I strongly believe that this should happen within the SDK, not here as it's an unexpected side-effect an SDK might not know nothing about. I'm not even sure whether it's even needed to disable the subscriptions here?

Reply:
Well - if that's the case I'm worried of the fact that they'll stay registered while they really shouldn't, ultimately resulting in double the subscriptions when you log in with another user (and this would keep going on as many times you `connect`/`disconnect`). Whenever the list of `threads` changes within the `ThreadManager`, we anyway unsubscribe from the specific `thread` listeners - so it's probably a good idea to flush everything out on the `ThreadManager` level as well.

But you do have a good point of some unexpected side-effects, anything specific you have in mind ?